warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java

```diff
@@ -377,6 +380,29 @@ private void serializeQuery(IteratorSetting newIteratorSetting) {
         newIteratorSetting.addOption(QueryOptions.MAX_PIPELINE_CACHED_RESULTS, "1");
     }
 
+    /**
+     * In the case of a document range there is no reason to spin up a {@link UniqueTransform}
+     *
+     * @param newIteratorSetting
+     *            the IteratorSettings
+     * @param ranges
+     *            the collection of ranges to search
+     */
+    protected void pruneUniqueOptions(IteratorSetting newIteratorSetting, Collection<Range> ranges) {
+        if (ranges.size() > 1) {
+            // Do not attempt if more than one range was provided
+            return;
+        }
+
+        Range range = ranges.iterator().next();
+        if (range.getStartKey().getColumnFamily().getLength() > 0 && range.isStartKeyInclusive()) {
+            // we have a document range
+            newIteratorSetting.removeOption(QueryOptions.UNIQUE_FIELDS);
+            newIteratorSetting.removeOption(QueryOptions.UNIQUE_CACHE_BUFFER_SIZE);
+            newIteratorSetting.removeOption(QueryOptions.MOST_RECENT_UNIQUE);
+        }
+    }
+
```

Comment:
Is it possible to leverage this option in a unit test or is it more practical to check it with integration testing on system?

Reply:
I can absolutely add a unit test for this